### PR TITLE
kie-kogito-docs-664: Create a guide for the SonataFlow Operator driven Supporting Services Knative Eventing system configuration

### PR DESCRIPTION
**Pull request by wmedvede**
Closes:  #664 

**Description:**

Please make sure that your PR meets the following requirements:

- [X] You have read the [contributions doc](https://github.com/apache/incubator-kie-kogito-docs/blob/main/CONTRIBUTING.md)
- [X] Pull Request title is properly formatted: `Issue-XYZ Subject`
- [X] Pull Request title contains the target branch if not targeting main: `[0.9.x] Issue-XYZ Subject`
- [X] The nav.adoc file has a link to this guide in the proper category
- [X] The index.adoc file has a card to this guide in the proper category, with a meaningful description

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 1]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
 For example, `sonataflow-platform-example-data-index-service`.
 ====
 
+== Configuring the Supporting Services Eventing System
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:183]:
```suggestion
== Configuring the supporting services Eventing system
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 2]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+The {operator_name} is designed to use the link:{knative_eventing_url}[Knative Eventing] system to resolve all the event communication between these services.
+
+[NOTE]
+====
+In a regular {product_name} installation, the preferred method is to use the <<platform-scoped-eventing-system-configuration, Platform scoped Eventing System Configuration>>, while the <<service-scoped-eventing-system-configuration, Service scoped Eventing System configuration>> is reserved only for advanced use cases.
+====
+
+[#platform-scoped-eventing-system-configuration]
+=== Platform scoped Eventing System Configuration
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:199]:
```suggestion
=== Platform-scoped Eventing system configuration
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 3]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+To configure a platform scoped eventing system, you must use the field `spec.eventing.broker.ref` in the `SonataFlowPlatform` CR to refer a Knative Eventing Broker.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:201]:
```suggestion
To configure a platform-scoped eventing system, you must use the field `spec.eventing.broker.ref` in the `SonataFlowPlatform` CR to refer to a Knative Eventing Broker.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 4]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+    broker:
+      ref:
+        name: example-broker <1>
+        namespace: example-broker-namespace <2>
+        apiVersion: eventing.knative.dev/v1
+        kind: Broker
+----
+
+<1> Name of the Knative Eventing Broker.
+<2> Optional: Defines the namespace of the Knative Eventing Broker. Defaults to the SonataFlowPlatform namespace. In general, we recommend to create the Knative Eventing Broker in the same namespace as the SonataFlowPlatform.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:230]:
```suggestion
<2> Optional: Defines the namespace of the Knative Eventing Broker. Defaults to the SonataFlowPlatform namespace. In general, we recommend creating the Knative Eventing Broker in the same namespace as the SonataFlowPlatform.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 5]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+[NOTE]
+====
+In production environments, you must use a production-ready broker, like the link:{knative_eventing_kafka_broker_url}[Knative Kafka Broker].
+====
+
+[#service-scoped-eventing-system-configuration]
+=== Service scoped Eventing System configuration
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:238]:
```suggestion
=== Service scoped Eventing system configuration
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 6]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+A service scoped eventing system configuration provides the ability to do a fine-grained configuration of the eventing system for the <<data-index-eventing-system-configuration, Data Index>> or the <<jos-service-eventing-system-configuration, Jobs Service>>.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:240]:
```suggestion
A service scoped eventing system configuration provides the ability to do a fine-grained configuration of the Eventing system for the <<data-index-eventing-system-configuration, Data Index>> or the <<jos-service-eventing-system-configuration, Jobs Service>>.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 7]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+[NOTE]
+====
+In a regular {product_name} installation, the preferred method is to use a <<platform-scoped-eventing-system-configuration, Platform scoped Eventing System Configuration>>, while the service scoped configuration is reserved only for advanced use cases.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:244]:
```suggestion
In a regular {product_name} installation, the preferred method is to use a <<platform-scoped-eventing-system-configuration, Platform-scoped Eventing System Configuration>>, while the service-scoped configuration is reserved only for advanced use cases.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 8]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+====
+
+[#data-index-eventing-system-configuration]
+=== Data Index Eventing System configuration
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:248]:
```suggestion
=== Data Index Eventing system configuration
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 9]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+To configure a service scoped eventing system for the Data Index, you must use the field `spec.services.dataIndex.source.ref` in the `SonataFlowPlatform` CR to refer a specific Knative Eventing Broker.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:250]:
```suggestion
To configure a service-scoped eventing system for the Data Index, you must use the field `spec.services.dataIndex.source.ref` in the `SonataFlowPlatform` CR to refer to a specific Knative Eventing Broker.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 10]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+      source:
+        ref:
+          name: data-index-source-example-broker <1>
+          namespace: data-index-source-example-broker-namespace <2>
+          apiVersion: eventing.knative.dev/v1
+          kind: Broker
+----
+
+<1> Name of the Knative Eventing Broker to `consume` events from.
+<2> Optional: Defines the namespace of the Knative Eventing Broker. Defaults to the SonataFlowPlatform namespace. In general, we recommend to create the Knative Eventing Broker in the same namespace as the SonataFlowPlatform.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:273]:
```suggestion
<2> Optional: Defines the namespace of the Knative Eventing Broker. Defaults to the SonataFlowPlatform namespace. In general, we recommend creating the Knative Eventing Broker in the same namespace as the SonataFlowPlatform.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 11]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+[NOTE]
+====
+In production environments, you must use a production-ready broker, like the link:{knative_eventing_kafka_broker_url}[Knative Kafka Broker].
+====
+
+[#jos-service-eventing-system-configuration]
+=== Jobs Service Eventing System configuration
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:281]:
```suggestion
=== Jobs Service Eventing system configuration
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 12]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+To configure a service scoped eventing system for the Jobs Service, you must use the fields `spec.services.jobService.source.ref` and `spec.services.jobService.sink.ref` in the `SonataFlowPlatform` CR.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:283]:
```suggestion
To configure a service-scoped eventing system for the Jobs Service, you must use the fields `spec.services.jobService.source.ref` and `spec.services.jobService.sink.ref` in the `SonataFlowPlatform` CR.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 13]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+This information signals the {operator_name} to automatically link the Jobs Service to `consume` and `produce` the {product_name} system events from that configurations respectively.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:285]:
```suggestion
This information signals the {operator_name} to automatically link the Jobs Service to `consume` and `produce` the {product_name} system events from that configuration respectively.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 14]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+      sink:
+        ref:
+          name: jobs-service-sink-example-broker <3>
+          namespace: jobs-service-sink-example-broker-namespace <4>
+          apiVersion: eventing.knative.dev/v1
+          kind: Broker
+----
+
+<1> Name of the Knative Eventing Broker to `consume` events from.
+<2> Optional: Defines the namespace of the Knative Eventing Broker. Defaults to the SonataFlowPlatform namespace. In general, we recommend to create the Knative Eventing Broker in the same namespace as the SonataFlowPlatform.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:312]:
```suggestion
<2> Optional: Defines the namespace of the Knative Eventing Broker. Defaults to the SonataFlowPlatform namespace. In general, we recommend creating the Knative Eventing Broker in the same namespace as the SonataFlowPlatform.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 15]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+<3> Name of the Knative Eventing Broker to `produce` events on.
+<4> Optional: Defines the namespace of the Knative Eventing Broker. Defaults to the SonataFlowPlatform namespace. In general, we recommend to create the Knative Eventing Broker in the same namespace as the SonataFlowPlatform.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:314]:
```suggestion
<4> Optional: Defines the namespace of the Knative Eventing Broker. Defaults to the SonataFlowPlatform namespace. In general, we recommend creating the Knative Eventing Broker in the same namespace as the SonataFlowPlatform.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 16]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+[NOTE]
+====
+In production environments, you must use production-ready brokers, like the link:{knative_eventing_kafka_broker_url}[Knative Kafka Broker].
+====
+
+[#cluster-scoped-eventing-system-configuration]
+=== Cluster scoped Eventing System configuration
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:322]:
```suggestion
=== Cluster-scoped Eventing system configuration
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 17]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+When you use a <<cluster-scoped-deployment>> deployment, the supporting services are automatically linked to the `Broker` configured in the `SonataFlowPlatform` CR referred by the `SonataFlowClusterPlatform` CR.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:324]:
```suggestion
When you use a <<cluster-scoped-deployment>> deployment, the supporting services are automatically linked to the `Broker` configured in the `SonataFlowPlatform` CR referred to by the `SonataFlowClusterPlatform` CR.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 18]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+=== Eventing System configuration precedence rules
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:326]:
```suggestion
=== Eventing system configuration precedence rules
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 19]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+To configure the eventing system for a supporting service, the {operator_name} use the following precedence rules:
+
+* If the supporting service has a configured eventing system, by using any of the <<data-index-eventing-system-configuration>> or <<jos-service-eventing-system-configuration>> respectively, that configuration apply.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:330]:
```suggestion
* If the supporting service has a configured eventing system, by using any of the <<data-index-eventing-system-configuration>> or <<jos-service-eventing-system-configuration>> respectively, that configuration applies.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 20]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+* If the `SonataFlowPlatform` CR enclosing the supporting service, is configured with a <<platform-scoped-eventing-system-configuration>>, that configuration apply.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:332]:
```suggestion
* If the `SonataFlowPlatform` CR enclosing the supporting service, is configured with a <<platform-scoped-eventing-system-configuration>>, that configuration applies.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 21]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+* If the current cluster, is configured with a <<cluster-scoped-eventing-system-configuration>>, that configuration apply.
+
+* If none of the previous configurations exists, the supporting service is configured to produce direct Http calls to deliver events.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:336]:
```suggestion
* If none of the previous configurations exists, the supporting service is configured to produce direct HTTP calls to deliver events.
```

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc [hunk 22]
```diff
@@ -180,6 +180,241 @@ When you use the common PostgreSQL configuration, the database schema for each s
...
+
+=== Eventing System linking objects
+
+The linking of the supporting services with the Eventing System is produced by using Knative Eventing SinkBindings and Triggers.
+These objects are automatically created by the {operator_name}, and facilitate the events production and consumption from the supporting services.
+
+The example below shows the Knative Native eventing objects created for the following `SonataFlowPlatform` CR.
```
Comment (kaldesai) [inline serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:343]:
```suggestion
The following example shows the Knative Native eventing objects created for the `SonataFlowPlatform` CR:
```

**Review by kaldesai — APPROVED** [review]
@wmedvede Just minor suggestions! Otherwise the content LGTM :)

File: serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc (hunk 6 above)
Comment (wmedvede) [reply serverlessworkflow/modules/ROOT/pages/cloud/operator/supporting-services.adoc:240]: @kaldesai some times, I see you suggested to use "service-scoped", in this paragraph, we keep "service scoped".
Missing suggestion? or we have a rational?

**Review by ricardozanini — APPROVED** [review]

**Pull request merged**